geopandas/tests/test_explore.py

```diff
@@ -309,6 +309,33 @@ def test_style_kwds(self):
         m = self.world.explore(column="pop_est", style_kwds=dict(color="black"))
         assert '"color":"black"' in self._fetch_map_string(m)
 
+        # custom style_function - geopandas/issues/2350
+        m = self.world.explore(
+            style_kwds={
+                "style_function": lambda x: {
+                    "fillColor": "red"
+                    if x["properties"]["gdp_md_est"] < 10 ** 6
+                    else "green",
+                    "color": "black"
+                    if x["properties"]["gdp_md_est"] < 10 ** 6
+                    else "white",
+                }
+            }
+        )
+        # two lines with formatting instructions from style_function.
+        # make sure each passes test
+        assert all(
+            [
+                ('"fillColor":"green"' in t and '"color":"white"' in t)
+                or ('"fillColor":"red"' in t and '"color":"black"' in t)
+                for t in [
+                    "".join(line.split())
+                    for line in m._parent.render().split("\n")
+                    if "return" in line and "color" in line
+                ]
+            ]
+        )
```

Comment:
Just a minor suggestion (feel free to ignore). We normally use this pattern in here


```py
        out_str = self._fetch_map_string(m)
        assert "target_string" in out_str
```

While your code correctly checks what it should check, I find it a bit unnecessarily complex to read. Would you mind simplifying it?

Reply:
it's months since I wrote this test case.  from what I recall `self._fetch_map_string(m)` over compresses the HTML/javascript so the conditions can not reliably be re-extracted from the compressed string.  I did attempt to build test case with that helper function but failed...   I am a big fan of follow existing in place standards rather than introduce additional approaches...

Reply:
That is probably true. Feel free to keep it as it is.